app/assets/javascripts/discourse/app/components/sidebar/categories-form-modal.hbs

```diff
@@ -1,75 +1,102 @@
+            style={{html-safe (border-color categories.0.color "left")}}
+          >
+
+            {{#each categories as |category|}}
+              <div
+                class="sidebar-categories-form__category-row"
+                data-category-id={{category.id}}
+                data-category-level={{category.level}}
+              >
+                <label
+                  class="sidebar-categories-form__category-label"
+                  for={{concat "sidebar-categories-form__input--" category.id}}
+                >
+                  <div class="sidebar-categories-form__category-badge">
+                    {{category-badge category}}
                   </div>
-                {{/unless}}
-              </label>
 
-              <Input
-                id={{concat "sidebar-categories-form__input--" category.id}}
-                class="sidebar-categories-form__input"
-                @type="checkbox"
-                @checked={{includes
-                  this.selectedSidebarCategoryIds
-                  category.id
-                }}
-                {{on "click" (action "toggleCategory" category.id)}}
-              />
-            </div>
-          {{/each}}
+                  {{#unless category.parentCategory}}
+                    <div class="sidebar-categories-form__category-description">
+                      {{dir-span category.description_excerpt htmlSafe="true"}}
+                    </div>
+                  {{/unless}}
+                </label>
+
+                <Input
+                  id={{concat "sidebar-categories-form__input--" category.id}}
+                  class="sidebar-categories-form__input"
+                  @type="checkbox"
+                  @checked={{includes
+                    this.selectedSidebarCategoryIds
+                    category.id
+                  }}
+                  {{on "click" (action "toggleCategory" category.id)}}
+                />
+              </div>
+            {{/each}}
+          </div>
+        {{/each}}
+      {{else}}
+        <div class="sidebar-categories-form__no-categories">
+          {{i18n "sidebar.categories_form_modal.no_categories"}}
         </div>
-      {{/each}}
-    {{else}}
-      <div class="sidebar-categories-form__no-categories">
-        {{i18n "sidebar.categories_form.no_categories"}}
-      </div>
-    {{/if}}
-  </form>
-</DModalBody>
+      {{/if}}
+    </form>
+  </DModalBody>
+
+  <div class="modal-footer sidebar-categories-form__modal-footer">
+    <DButton
+      @class="btn-primary sidebar-categories-form__save-button"
+      @label="sidebar.categories_form_modal.save"
+      @disabled={{this.saving}}
+      @action={{this.save}}
+    />
 
-<div class="modal-footer">
-  <DButton
-    @class="btn-primary sidebar-categories-form__save-button"
-    @label="sidebar.categories_form.save"
-    @disabled={{this.saving}}
-    @action={{this.save}}
-  />
+    {{#if (gt this.siteSettings.default_navigation_menu_categories.length 0)}}
```

Comment:
Just idea, shall we display reset to default button only if current selection is different than default?

Reply:
I thought about this but I find buttons that disappear and reappear very confusing since it causes something to flash on the screen.